### DISCUSSION TITLE
Please explain the syntax of this 'if' statement

### DISCUSSION THREAD

**Discussion by DavidEwan** [community/community#33684] [New to GitHub]
I'm looking at this page: 
https://docs.github.com/en/actions/learn-github-actions/contexts

and this statement in particular: 
if [[ ${{ steps.generate_number.outputs.random_number }} == 0 ]]; then exit 0; else exit 1; fi

Why are there two square brackets? What purpose do they serve and why are there two of them??
If you could point me to a docs page that explains that would be great. Thanks.

**Comment by Shivansh-Mittal** [comment]
I searched on google for this query and found this discussion: 
https://unix.stackexchange.com/questions/99185/what-do-square-brackets-mean-without-the-if-on-the-left

I hope this may help.

**Comment by thuongtruong1009** [comment]
Hi @DavidEwan 
Because this is the basic syntax standard set by ***GitHub***, to avoid the syntax overlap with [***template_literals***](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) in some languages like ***javascript*** ``` `${something}` ```
So it is a common standard to set and call variable names in some commands like Github Action
See Two square brackets syntax in [Github Action example syntax](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_calloutputs)

**Comment by airtower-luna** [comment]
That line is part of a `run` step running in Bash. The double square brackets are part of Bash's syntax for conditional expressions, and *double* square brackets support some extensions (e.g. regular expression matching) relative to the *single* square bracket conditionals. See [Conditional Constructs](https://www.gnu.org/software/bash/manual/html_node/Conditional-Constructs.html) in the Bash Manual for details.
